rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -1553,25 +1549,38 @@ protected void doCALL(){
 
         program.memoryExpand(outDataOffs, outDataSize);
 
-        MessageCall msg = new MessageCall(
+        return new MessageCall(
                 MsgType.fromOpcode(op),
                 DataWord.valueOf(calleeGas), codeAddress, value, inDataOffs, inDataSize,
                 outDataOffs, outDataSize);
+    }
 
-        callToAddress(codeAddress, msg);
-
-        program.step();
+    private DataWord calculateCallValue(ActivationConfig.ForBlock activations) {
+        DataWord value;
+        if (activations.isActive(RSKIP103)) {
+            // value is always zero in a DELEGATECALL or STATICCALL operation
+            value = op == OpCode.DELEGATECALL || op == OpCode.STATICCALL ? DataWord.ZERO : program.stackPop();
+        } else {
+            // value is always zero in a DELEGATECALL operation
+            value = op == OpCode.DELEGATECALL ? DataWord.ZERO : program.stackPop();
+        }
+        return value;
     }
 
-    private void callToAddress(DataWord codeAddress, MessageCall msg) {
-        ActivationConfig.ForBlock activations = program.getActivations();
-        PrecompiledContracts.PrecompiledContract contract = precompiledContracts.getContractForAddress(activations, codeAddress);
+    private long calculateGetMinimumTransferGas(DataWord value, long remainingGas) {
```

Comment:
This function is not cohesive and the naming is unclear. It's doing two things, and one of them is validating and maybe throwing an exception. At the very least is should be called calculateAndValidateMinimumTransferGas